units/host_monitoring_external_termination.go

```diff
@@ -125,45 +128,32 @@ func handleExternallyTerminatedHost(ctx context.Context, id string, env evergree
 			return false, errors.Wrapf(h.MarkReachable(), "error updating reachability for host %s", h.Id)
 		}
 		return false, nil
-	case cloud.StatusTerminated:
-		grip.Info(message.Fields{
-			"op_id":   id,
-			"message": "host terminated externally",
-			"host_id": h.Id,
-			"distro":  h.Distro.Id,
-		})
+	case cloud.StatusStopped, cloud.StatusTerminated:
+		// Avoid accidentally terminating non-agent hosts that are stopped (e.g.
+		// spawn hosts).
+		if cloudStatus == cloud.StatusStopped && (h.UserHost || h.StartedBy != evergreen.User) {
```

Comment:
why do we need to check `h.UserHost` (given `h.StartedBy != evergreen.User`)?

Reply:
This is just an excessive cautionary measure on my part. In the current state of things, hosts always have these characteristics:
1. An agent host is started by "mci" and is _never_ a user-owned host.
2. All other host types are started by a user other than "mci" and are always user-owned hosts.

I just wanted to have a little extra future-proofing proection just in case one of those things stops being the case (e.g. introducing some new user-owned host that is started by "mci" for whatever reason), since accidentally deleting someone's host would be really bad.